x-pack/plugins/spaces/public/space_list/space_list_internal.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, ReactNode, useEffect } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiBadge } from '@elastic/eui';
+import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { EuiToolTip } from '@elastic/eui';
+import { EuiButtonEmpty } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import type { SpaceListProps } from '../../../../../src/plugins/spaces_oss/public';
+import { ShareToSpacesData, ShareToSpaceTarget } from '../types';
+import { ALL_SPACES_ID, UNKNOWN_SPACE } from '../../common/constants';
+import { useSpaces } from '../spaces_context';
+import { SpaceAvatar } from '../space_avatar';
+
+const DEFAULT_DISPLAY_LIMIT = 5;
+
+export const SpaceListInternal = ({
+  namespaces,
+  displayLimit = DEFAULT_DISPLAY_LIMIT,
+  behaviorContext,
+}: SpaceListProps) => {
+  const { shareToSpacesDataPromise } = useSpaces();
+
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [shareToSpacesData, setShareToSpacesData] = useState<ShareToSpacesData>();
+
+  useEffect(() => {
+    shareToSpacesDataPromise.then((x) => {
+      setShareToSpacesData(x);
+    });
+  }, [shareToSpacesDataPromise]);
+
+  if (!shareToSpacesData) {
+    return null;
+  }
+
+  const isSharedToAllSpaces = namespaces?.includes(ALL_SPACES_ID);
+  const unauthorizedCount = (namespaces?.filter((namespace) => namespace === UNKNOWN_SPACE) ?? [])
```

Comment:
__nit:__ type definition says `namespaces` should always be an array, do we really need `?` here, if so then we need to update type definition:
```suggestion
  const isSharedToAllSpaces = namespaces.includes(ALL_SPACES_ID);
  const unauthorizedCount = (namespaces.filter((namespace) => namespace === UNKNOWN_SPACE) ?? [])
```

Reply:
No, `namespaces` used to be optional but now it's required (technically a saved object doesn't have to have this field, but all shareable objects do, so it made sense to change this to a required property). Forgot to remove this optional chaining when I made that change, good catch 😄